.github/workflows/publish.yml

```diff
@@ -1,21 +1,55 @@
 name: "publish"
-on: workflow_dispatch
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - "*"
 
 jobs:
   publish-tauri:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-20.04
-          - macos-11
-          - windows-2022
+        build:
+          # for ubuntu with arch, we need more testing with cross compilation libraries
+          # since releasing on arm-linux is not a priority, we skipped it for now
+          - os: ubuntu-20.04
+            target: x86_64-unknown-linux-gnu
+            suffix: ubuntu-x86_64
+            rustflags: ""
+          - os: macos-12
+            target: x86_64-apple-darwin
+            suffix: macos-x86_64
+            rustflags: ""
+          - os: macos-12
+            target: aarch64-apple-darwin
+            suffix: macos-aarch64
+            rustflags: ""
+          - os: windows-2022
+            target: x86_64-pc-windows-msvc
+            suffix: windows-x86_64
+            rustflags: ""
+
+    runs-on: ${{ matrix.build.os }}
+
+    env:
+      PRODUCTION_TARGET: src-tauri/target/${{ matrix.build.target }}/release/bundle
+      RUSTFLAGS: ${{ matrix.build.rustflags }}
 
-    runs-on: ${{ matrix.os }}
     steps:
+      - name: Maximize build space
+        if: runner.os == 'Linux'
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
       - name: git checkout
         uses: actions/checkout@v2
 
+      - name: Extract version
+        id: extract_version
+        uses: Saionaro/extract-package-version@v1.0.6
```

Comment:
Wouldn't it be better to use tag as a version instead? It will be the same in most cases, but you could also do nightly builds or other interesting things and still have executabled being called uniquely instead of all look the same. Just a thought.

Reply:
yes this was a remnant of the previous strategy for uploading the executables to assets, will change

Reply:
Oh actually no, just remembered:
when tauri builds the executable, it automatically appends the version string to the executable name. So we have to get the version from the package.json for this workflow. I can't remove that.

However, if your suggestion was not to get rid of this, but only to use tag as the version, I can do that

Reply:
Hm, I guess we can use tag for the final name of the file and leave whatever Tauri generates as source. BTW, for file copying you can use `*` and not exact match, thus potentially avoiding bothering with specific versions.

Reply:
There are other files in the directory that I'm copying the bundle, `*` did not seem safe to me... 
This extra step is very minimal and does not take much time. I believe it makes the workflow safer (we have access to the full filename)

Reply:
Sure. But you don't have to use just `*`, you can use `target/*/*.msi` for instance and capture file `target/x86-64.../something.msi`. Anyway, works fine as is too.